plugin/src/test/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfiguratorTest.java

```diff
@@ -90,6 +92,22 @@ public void export_mapping_should_not_be_null() throws Exception {
         assertEquals("plainText", configNode.getScalarValue("markupFormatter"));
     }
 
+    @Test
+    public void shouldThrowConfiguratorException() {
+        Mapping config = new Mapping();
+        config.put("fooo", "foo");
+        config.put("barr", "true");
+        config.put("qixz", "123");
+        ConfiguratorRegistry registry = ConfiguratorRegistry.get();
+        try {
+            registry.lookupOrFail(Foo.class).configure(config, new ConfigurationContext(registry));
+            fail("above action is excepted to throw ConfiguratorException!");
+        } catch (ConfiguratorException e) {
+            assertThat(e.getMessage(), is("Invalid configuration elements for type class io.jenkins.plugins.casc.impl.configurators.DataBoundConfiguratorTest$Foo : barr,fooo,qixz.\n"
+                + "Available attributes : bar, foo, other, qix, zot"));
```

Comment:
isn't this testing BaseConfigurator, not DataboundConfigurator?

Reply:
Your correct, I have been unable to get DataboundConfigurator to throw the exception.

Reply:
@casz yeah exactly even I tried the approach of `getInstance` looking for the configure function to complain but it did not :(

Reply:
@timja perhaps you want to take a gander at it?

Reply:
You need to make the constructor throw an exception,
```
            if (qix == 99) {
                throw new NullPointerException("Magic test fail");
            }
```

the clue was that this was reported by a user of the ec2-plugin which had weird stuff in its constructor

```
io.jenkins.plugins.casc.ConfiguratorException: foo: Failed to construct instance of class io.jenkins.plugins.casc.impl.configurators.DataBoundConfiguratorTest$Foo.
 Constructor: public io.jenkins.plugins.casc.impl.configurators.DataBoundConfiguratorTest$Foo(java.lang.String,boolean,int).
 Arguments: [java.lang.String, java.lang.Boolean, java.lang.Integer].
 Expected Names of Parameters: foo, bar, qix.
 Expected Type of Parameters: java.lang.String, boolean, int
```

Reply:
is it worth joining the parameters so it looks like:

```
[ foo java.lang.String, bar java.lang.Boolean, qix java.lang.Integer ]
```

Reply:
@timja  Throwing the null pointer exception still does not ensure that the `FailedConstructorException` gets thrown.It is still reverting to the `InvalidConfigurationException`.

Reply:
you need to change the parameters back to being valid
```
        Mapping config = new Mapping();
        config.put("foo", "foo");
        config.put("bar", "abcd");
        config.put("qix", "99");
```

Reply:
Did that , still the same :(

Reply:
I've pushed the code, not sure why it's not working for you

Reply:
Combined the params.Kindly review